04/app.js

```diff
@@ -0,0 +1,28 @@
+const divsList = document.querySelectorAll('div');
+
+const removeClass = function (arr, className) {
+    return arr.forEach(el => {
+        el.classList.remove(className);
+    });
+}
+
+const addClassName = function (e) {
+    if (e.target.className !== 'child') {
+        this.classList.add('clicked');
+        e.stopPropagation();
+    } else {
+        removeClass(divsList, 'clicked');
```

Comment:
Funkcja nazywa sie "dodaj klasę", a w jej wnętrzu jest usuwanie. Trochę to dziwnie wygląda ;P
Może warto wybrać inną drogę do rozwiązania, aby była bardziej czytelna?
Czy faktycznie trzeba tu coś usuwać lub zatrzymywać propagację?

Reply:
poprawione - wcześniej trochę inaczej podszedłem do tego zadania, tylko kliknięcie w środkowy `div` miało rozpocząć wypełnianie kolejnych, a kliknięcie w inny element miało wypełniać tylko kliknięty :)